app/models/grda_warehouse/auth_policies/legacy_user_role_policy.rb

```diff
@@ -6,16 +6,24 @@
 
 # TODO: START_ACL remove after ACL migration is complete
 class GrdaWarehouse::AuthPolicies::LegacyUserRolePolicy
+  include Memery
   attr_reader :user
 
-  def initialize(user:)
+  def initialize(user:, access_group_ids:)
     @user = user
+    @access_group_ids = access_group_ids
   end
 
   Role.permissions.each do |permission|
     method_name = :"#{permission}?"
     define_method method_name do
-      !!user.public_send(method_name)
+      # check if the user has permission on any role
+      return false unless user.public_send(method_name)
+
+      # check if the user in in any of the access groups
+      user.access_groups.where(id: @access_group_ids).exists?
```

Comment:
As discussed, this probably needs to account for the various ROI setups.  Proposal is to generate a table of Client ROI that we can more easily union with the project-based permissions so we can accommodate the ROI setup.